### ISSUE TITLE
Bug: (no-throw-literal has a bug)

### ISSUE THREAD

**Issue by Silentdoer** [eslint/eslint#17803]
### Environment

Node version: 16.13.1
npm version: 8.1
Local ESLint version: 8.0
Global ESLint version: 8.0
Operating System: win10

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
"no-throw-literal": "error"
```
</details>

```js
function fff(name: string): void {
  if (name === "foo") {
    // this should be error, because name is not Error type
    throw name
  }
}
```

### What did you expect to happen?

throw just Error type

### What actually happened?

throw can be string type

### Link to Minimal Reproducible Example

http://silentdoer.com/nonproj

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Tanujkanti4441** [comment]
Hi @Silentdoer, thanks for the issue

> function fff(name: string): void {
  if (name === "foo") {
    // this should be error, because name is not Error type
    throw name
  }
}

i think this example is similar to the first example of [Know Limitation](https://eslint.org/docs/latest/rules/no-throw-literal#known-limitations) section of this rule's docs that mentions it is the expected behaviour or am i missing something? and also
 > http://silentdoer.com/nonproj

can you check this link, not able open it.

**Comment by mdjermanovic** [comment]
Yes, core rules don't have type info, so `no-throw-literal` cannot determine that this is a string value and thus not an Error object. A similar example already exists in the [Known Limitations](https://eslint.org/docs/latest/rules/no-throw-literal#known-limitations) section for this rule.

You can use the [`@typescript-eslint/no-throw-literal`](https://typescript-eslint.io/rules/no-throw-literal/) rule instead of the core rule to catch cases like this. Here's [typescript-eslint demo](https://typescript-eslint.io/play/#ts=5.3.2&fileType=.ts&code=PQKgpgzgNglgdgFwAQAEEE8AOkDGAnGTBAWklkWDgHtiEALPKgd2NgTDwEMoAuJAIg6M8-EMABQ4gGYBXODgQwqcJFLUAKOJwC2YPhAQE4AcwCUfAG5UYAEyQBvcUiQwpSTTrBIAvL4FSqKn5TBydnJGBgJHoYCCQIOioZKDsAIy8hKjwAGiR0nE4ZCC8tXRc46mQAUTxhaKwwMOd6RiYkUsbnAF9xLqA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaAFwAtp8B3Q%2BAS2MWgEN50p8AzDyAGnG0gABYgE8ADigDG0WmOKEUdXMQD0BEuSo16jFm0iMK0SPwC%2BIU0A&tsconfig=&tokens=false) with the original example.
